tests/python/contrib/test_ethosu/test_codegen.py

```diff
@@ -48,122 +47,147 @@ def get_shape_expr(in_expr, out_expr):
     return shape
 
 
-@pytest.mark.parametrize(
-    "accel_type",
-    ACCEL_TYPES,
-)
-def test_ethosu_conv2d(accel_type):
-    def create_graph_single(input_tensor_name, input_tensor_shape, input_tensor_dtype):
-        c1_params = relay_ir_builder.QnnConv2DParams(input_tensor_dtype)
-        c1_params.ifm.shape = input_tensor_shape
-        c1_params.kernel.shape = (3, 3, c1_params.ifm.shape[3], 32)
-        c1_params.kernel.sc = relay.const(np.random.rand(32) * 2, "float32")
-        c1_params.strides = (1, 1)
-        c1_params.pad = "VALID"
-        c1_params.update_output_qnn_params(
-            input_tensor_dtype, input_tensor_dtype, input_tensor_dtype
-        )
-        input0 = relay.var(input_tensor_name, shape=c1_params.ifm.shape, dtype=c1_params.ifm.dtype)
-        c1, new_params = relay_ir_builder.create_qnn_conv2d(c1_params, input0)
-        c1_params.ofm.shape = get_shape_expr(input0, c1)
+@pytest.mark.parametrize("ifm_shape", [(1, 299, 299, 3), (1, 55, 55, 3)])
+@pytest.mark.parametrize("kernel_shape", [(3, 2, 3, 3), (1, 3, 3, 3)])
```

Comment:
Since in the rest of the NPU code "kernel shape" refers to the 2 dimensional height and width of the weights/kernel, I think it would be good to follow the same convention here, especially since it is not obvious from the 4D array what each number represents. I suggest you pass in the kernel shape as a 2D array and either have separate parameters for input/output channels or define them in the test body and use the same input and output channels for each test

Reply:
ack! for simplicity sake I'll use the same channels across all tests